### DISCUSSION TITLE
Any interest and plan to make a pwa app for Trilinum note

### DISCUSSION THREAD

**Discussion by kangaroolab** [TriliumNext/Trilium#5999] [General]
Love trilinum note and have been using it daily. Wonder any interest and plan to build a pwa version of it with limited functionality?

**Comment by shutaozhenzhen** [comment]
Doesn't trilium already have pwa functionality?

**Comment by kangaroolab** [comment]
it doesn't have one yet...it is electron app, which is about 380MB on linux.

what if we have options to carve out the browser side and make it pwa compatible? any idea how to do it?

**Reply by shutaozhenzhen** [reply]
It has.

trilium is already  carve out the browser side. It has server only build and docker build, both are used by browser(electron it's just the browser that can use)

And I believe, in browser mode, trilium has a pwa

Anyway, I'm not a developer and don't use it very often (especially not on linux).
Perhaps someone more experienced can explain

**Reply by eliandoran** [reply]
@kangaroolab , @shutaozhenzhen is correct, there is support for Trilium Notes (and TriliumNext Notes) as well for web. This is done via the server application which is available for Linux, but also as a Docker container. 

There is also Trilium.cc which offers paid hosting, if you are not into self-hosting. 

Let me know if you need any more information.

**Reply by kangaroolab** [reply]
Thanks @eliandoran. Appreciate the info. By pwa, I was looking for standalone without server, just like how other pwa works. Is it possible?

**Reply by shutaozhenzhen** [reply]
> Thanks @eliandoran. Appreciate the info. By pwa, I was looking for standalone without server, just like how other pwa works. Is it possible?

I kind of like this idea, to turn trilium into a app without server. But I believe this will be hard.

**Comment by yry-dev** [comment]
The experience on mobile is kinda hard to use a pwa or even just a better responsive layout would be appreciated

**Reply by kotlasu** [reply]
Amen to that.
